scheduler/src/cook/mesos/mesos_mock.clj

```diff
@@ -209,19 +211,37 @@
 (defn complete-task!
   "Marks the task as complete in the state and updates the resources
    Returns the updated state"
-  [state task-id scheduler driver task-state]
-  (log/debug "Completing task" {:task-id task-id :state task-state :task (get-in state [:task-id->task task-id])})
-  (if-let [task (get-in state [:task-id->task task-id])]
-    (do
-      (.statusUpdate scheduler driver (mesos-type/->pb :TaskStatus
-                                                       {:task-id {:value task-id}
-                                                        :state task-state}))
-      (-> state
-          (update :task-id->task #(dissoc % task-id))
-          (update :task-id->completed-task #(assoc % task-id (assoc task :complete-time (t/now))))
-          (update-in [:slave-id->host (:value (:slave-id task)) :available-resources]
-                     #(combine-resources (:resources task) %))))
-    state))
+  ([state task-id scheduler driver]
+   (complete-task! state task-id scheduler driver nil))
+  ([state task-id scheduler driver task-state]
+   (log/debug "Completing task" {:task-id task-id :state task-state :task (get-in state [:task-id->task task-id])})
+   (if-let [task (get-in state [:task-id->task task-id])]
+     (do
+       (let [task->complete-status (get-in state [:config :task->complete-status]) 
+             task-state (or task-state (task->complete-status task))]
+         (.statusUpdate scheduler driver (mesos-type/->pb :TaskStatus
+                                                          {:task-id {:value task-id}
+                                                           :state task-state}))
+         (-> state
+             (update :task-id->task #(dissoc % task-id))
+             (update :task-id->completed-task #(assoc % task-id (assoc task :complete-time (t/now))))
```

Comment:
`complete-trigger-chan` chimes every 20ms and calls `complete-tasks!` that eventually calls `complete-task!`